### PR TITLE
feat(providers): add Tencent QQ Provider

### PR DESCRIPTION
**Pull request by HiChen404**
## ☕️ Reasoning

Refactor Tencent QQ Provider with TypeScript.

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

## 🎫 Affected issues

[Add Tencent QQ Provider](https://github.com/nextauthjs/next-auth/pull/4354)

Because I do not have permission to continue to commit, I opened a new PR.

### CODE REVIEW

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Please move this to https://github.com/nextauthjs/next-auth/tree/main/packages/core/src/providers

(Check out any provider there for instructions on formatting)

**Comment by HiChen404** [comment]
I modified the commit, please take a look. 

Thanks for taking the time to maintain this project. :D

**Comment by Tian-que** [comment]
Good job! Thank you for doing this

File: packages/core/src/providers/qq.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ *```js
+ * import Auth from "@auth/core"
+ * import TencentQQProvider from "next-auth/providers/qq";
+ *
+ * const request = new Request(origin)
+ * const response = await Auth(request, {
+ *   providers: [
+ *    TencentQQProvider({
+ *      clientId: process.env.TENCENT_ID,
+ *      clientSecret: process.env.TENCENT_APP_KEY,
```
Comment (ThangHuuVu) [inline packages/core/src/providers/qq.ts:43]:
```suggestion
 *      clientSecret: process.env.TENCENT_SECRET,
```

File: packages/core/src/providers/qq.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      url: "https://graph.qq.com/oauth2.0/me",
+      async request(context) {
+        const response = await fetch(
+          "https://graph.qq.com/oauth2.0/me?" +
+            new URLSearchParams({
+              access_token: context.tokens.access_token || "",
+              fmt: "json",
+            }),
+          {
+            method: "GET",
```
Comment (ThangHuuVu) [inline packages/core/src/providers/qq.ts:112]: since the default is `GET`, we can remove this options

File: packages/core/src/providers/qq.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        const openIDInfo = await response.json()
+        const userInfoResponse = await fetch(
+          "https://graph.qq.com/user/get_user_info?" +
+            new URLSearchParams({
+              access_token: context.tokens.access_token || "",
+              oauth_consumer_key: openIDInfo.client_id,
+              openid: openIDInfo.openid,
+            }),
+          {
+            method: "GET",
```
Comment (ThangHuuVu) [inline packages/core/src/providers/qq.ts:124]: since the default is `GET`, we can remove this options

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]
Thanks for the PR, I left some comments. Could you also add an entry of the provider at `.github/ISSUE_TEMPLATE/2_bug_provider.yml`? 🙏

**Comment by yiyungent** [comment]
@Tian-que 
Thank you very much. Considering the Chinese market, QQ login is absolutely necessary ! ! !

**Pull request closed without merging**